packages/react-router/src/CatchBoundary.tsx

```diff
@@ -9,13 +12,23 @@ export function CatchBoundary(props: {
   onCatch?: (error: Error, errorInfo: ErrorInfo) => void
 }) {
   const errorComponent = props.errorComponent ?? ErrorComponent
+  const router = useRouter()
 
   return (
     <CatchBoundaryImpl
       getResetKey={props.getResetKey}
       onCatch={props.onCatch}
       children={({ error, reset }) => {
         if (error) {
+          if (isRedirect(error)) {
+            const redirect = router.resolveRedirect({
+              ...error,
+              _fromLocation: router.state.location,
+            })
+            return (
+              <Navigate {...redirect} replace={true} ignoreBlocker={true} />
```

Comment:
What's the rationale for setting `replace=true` here? Is this something that's required to make it work?
Or is it purely for allowing the user to go from `1` to `-1` in a single back button click when considering the user being at position `1` in `[ -1 , 0 , 1 ]`?

Asking, because if this is not something required on our side to make it functional at runtime, then perhaps, we shouldn't be enforcing `replace=true` on our side forcing a history replacement. Maybe this should be something the user should opt-in to?